videostore/rawsegmenter.c

```diff
@@ -147,37 +128,69 @@ int video_store_raw_seg_init_h264(struct raw_seg_h264 **ppRS, // OUT
   return ret;
 }
 
-int video_store_raw_seg_write_h264_packet(struct raw_seg_h264 *rs,  // IN
-                                          const char *payload,      // IN
-                                          const size_t payloadSize, // IN
-                                          const int64_t pts,        // IN
-                                          const int isIdr           // IN
+int video_store_raw_seg_init_h264(struct raw_seg **ppRS,     // OUT
+                                  const int segmentSeconds,  // IN
+                                  const char *outputPattern, // IN
+                                  const int width,           // IN
+                                  const int height           // IN
+) {
+  const struct AVCodec *codec = avcodec_find_decoder(AV_CODEC_ID_H264);
```

Comment:
the previous code looked for an encoder... we don't need to do that we can instead look for a decoder (which is semantically what we need).